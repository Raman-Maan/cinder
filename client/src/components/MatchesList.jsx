import React, { Component } from 'react';
import './styles/MatchesList.css';

import MatchesListItem from '../components/MatchesListItem';

class MatchesList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      active: false
    };
  }

  onListItemClick(id) {
    this.setState({ active: id });
    this.props.clickHandler(id);
  }

  render() {
    if (!this.props.matches || !this.props.matches.length) {
      return (
        <div className="MatchesList">
          <div className="empty">
            <h3 className="msg align-self-center">
              No matches yet. Get swiping!
            </h3>
          </div>
        </div>
      );
    }

    const listItems = this.props.matches.map(
      ({ id, title, subtitle, img, date }) => (
        <MatchesListItem
          key={id}
          title={title}
          subtitle={subtitle}
          img={img}
          date={date}
          active={id === this.state.active}
          // eslint-disable-next-line react/jsx-no-bind
          onClick={this.onListItemClick.bind(this, id)}
        />
      )
    );

    return (
      <div className="MatchesList">
        <ul>{listItems}</ul>
      </div>
    );
  }
}

export default MatchesList;
