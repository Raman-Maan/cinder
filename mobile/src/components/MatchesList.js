import React from 'react';
import { connect } from 'react-redux';
import { List, ListItem, Text, Container, Spinner } from 'native-base';
import { StyleSheet } from 'react-native';

import { matchesFetchData } from '../actions';
import MatchesListItem from './MatchesListItem';
import { loggedInUser } from '../../env';


export class MatchesList extends React.Component {
  componentDidMount() {
    this.props.fetchData(loggedInUser);
  }

  render() {
    const msg = this.props.loading
      ? 'Loading...'
      : !this.props.matches.length
        ? 'No matches yet. Get swiping!'
        : 'There was an error loading matches';

    if (
      this.props.errored ||
      (this.props.loading && !this.props.matches.length) ||
      !this.props.matches.length
    ) {
      return (
        <Container style={styles.container}>
          <Text>{msg}</Text>
        </Container>
      );
    }

    const listItems = this.props.matches.map(({
      userID, userName, userBio, primaryPic, matchDate,
    }) =>
      (<MatchesListItem
        key={userID}
        userID={userID}
        userName={userName}
        userBio={userBio}
        primaryPic={primaryPic}
        matchDate={matchDate}
      />));

    return (
      <List>
        {listItems}
      </List>
    );
  }
}

const mapStateToProps = state => ({
  matches: state.matches,
  errored: state.matchesHasErrored,
  loading: state.matchesIsLoading,
});

const mapDispatchToProps = dispatch => ({
  fetchData: userID => dispatch(matchesFetchData(userID)),
});

export default connect(mapStateToProps, mapDispatchToProps)(MatchesList);


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
