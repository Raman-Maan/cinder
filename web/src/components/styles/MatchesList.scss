$bg-color: #ddd;
$border-color: #bbb;
$empty-msg-color: #aaa;

.MatchesList {
  width: 100%;
  height: 100%;
  max-height: 100vh;
  overflow-y: auto;

  background-color: $bg-color;
  border-right: 1px solid $border-color;

  .empty {
    display: flex;
    width: 100%;
    height: 100%;

    padding: 1em;

    justify-content: center;

    .msg {
      color: $empty-msg-color;
    }
  }

  ul {
    height: 100%;
    padding: 0;
    margin: 0;
  }
}
