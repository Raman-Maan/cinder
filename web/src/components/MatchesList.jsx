import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchAllMatches } from '../actions';
import PropTypes from 'prop-types';
import MatchesListItem from '../components/MatchesListItem';
import './styles/MatchesList.css';

export class MatchesList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      active: false
    };
  }

  componentDidMount() {
    this.props.fetchData(this.props.auth.userID, this.props.auth.token);
  }

  onListItemClick(id) {
    this.setState({ active: id });
    this.props.clickHandler(id);
  }

  render() {
    const msg = this.props.loading
      ? 'Loading...'
      : !this.props.matches.length
        ? 'No matches yet. Get swiping!'
        : 'There was an error loading matches';

    if (
      this.props.errored ||
      (this.props.loading && !this.props.matches.length) ||
      !this.props.matches.length
    ) {
      return (
        <div className="MatchesList">
          <div className="empty">
            <h3 className="msg align-self-center">{msg}</h3>
          </div>
        </div>
      );
    }

    const listItems = this.props.matches.map(
      ({ userID, userName, userBio, primaryPic, matchDate }) => (
        <MatchesListItem
          key={userID}
          title={userName}
          subtitle={userBio}
          img={primaryPic}
          date={matchDate}
          active={userID === this.state.active}
          // eslint-disable-next-line react/jsx-no-bind
          onClick={this.onListItemClick.bind(this, userID)}
        />
      )
    );

    return (
      <div className="MatchesList">
        <ul>{listItems}</ul>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  matches: state.match.matches,
  errored: state.match.errored,
  loading: state.match.loading,
  auth: state.auth
});

const mapDispatchToProps = dispatch => ({
  fetchData: (userID, token) => dispatch(fetchAllMatches(userID, token))
});

MatchesList.propTypes = {
  fetchData: PropTypes.func.isRequired,
  auth: PropTypes.shape({
    isFetching: PropTypes.bool,
    isAuthenticated: PropTypes.bool.isRequired,
    token: PropTypes.string,
    userID: PropTypes.string,
    message: PropTypes.string
  }),
  clickHandler: PropTypes.func.isRequired,
  matches: PropTypes.array,
  loading: PropTypes.bool,
  errored: PropTypes.bool
};

export default connect(mapStateToProps, mapDispatchToProps)(MatchesList);
