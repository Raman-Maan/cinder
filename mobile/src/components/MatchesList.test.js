import React from 'react';
import renderer from 'react-test-renderer';

import { MatchesList } from './MatchesList';

describe('<MatchesList />', () => {
  const testData = ['A', 'B', 'C', 'D', 'E'].map((x, index) => ({
    userID: index,
    userName: x,
    userBio: x,
    primaryPic: x,
    matchDate: new Date(`01/0${index + 1}/2018`),
  }));

  const mockFetch = jest.fn();

  it('renders when empty', () => {
    const rendered = renderer.create(<MatchesList matches={[]} fetchData={mockFetch} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders when loading', () => {
    const rendered = renderer.create(<MatchesList loading fetchData={mockFetch} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders when errored', () => {
    const rendered = renderer.create(<MatchesList errored fetchData={mockFetch} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders with matches', () => {
    const rendered = renderer.create(<MatchesList matches={testData} fetchData={mockFetch} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
