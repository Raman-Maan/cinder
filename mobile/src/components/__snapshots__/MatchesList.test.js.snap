// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchesList /> renders when empty 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    No matches yet. Get swiping!
  </Text>
</View>
`;

exports[`<MatchesList /> renders when errored 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    There was an error loading matches
  </Text>
</View>
`;

exports[`<MatchesList /> renders when loading 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    Loading...
  </Text>
</View>
`;

exports[`<MatchesList /> renders with matches 1`] = `
<View
  closeOnRowBeginSwipe={false}
  closeOnRowPress={true}
  closeOnScroll={true}
  directionalDistanceChangeThreshold={2}
  disableLeftSwipe={false}
  disableRightSwipe={false}
  leftOpenValue={0}
  previewFirstRow={false}
  recalculateHiddenLayout={false}
  rightOpenValue={0}
  style={Object {}}
  swipeToOpenPercent={50}
  virtual={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      avatar={true}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 15,
          "paddingBottom": null,
          "paddingRight": null,
          "paddingTop": null,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 0,
            "flexDirection": "row",
          }
        }
        virtual={undefined}
      >
        <Image
          source={
            Object {
              "uri": "A",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              },
              Object {
                "resizeMode": "contain",
              },
            ]
          }
          virtual={undefined}
        />
      </View>
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": null,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 1,
            "marginLeft": 15,
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          A
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          numberOfLines={1}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          A
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "stretch",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 0,
            "paddingBottom": 10,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          style={
            Object {
              "alignSelf": null,
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "200",
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          2018-01-01
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      avatar={true}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 15,
          "paddingBottom": null,
          "paddingRight": null,
          "paddingTop": null,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 0,
            "flexDirection": "row",
          }
        }
        virtual={undefined}
      >
        <Image
          source={
            Object {
              "uri": "B",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              },
              Object {
                "resizeMode": "contain",
              },
            ]
          }
          virtual={undefined}
        />
      </View>
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": null,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 1,
            "marginLeft": 15,
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          B
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          numberOfLines={1}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          B
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "stretch",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 0,
            "paddingBottom": 10,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          style={
            Object {
              "alignSelf": null,
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "200",
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          2018-01-02
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      avatar={true}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 15,
          "paddingBottom": null,
          "paddingRight": null,
          "paddingTop": null,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 0,
            "flexDirection": "row",
          }
        }
        virtual={undefined}
      >
        <Image
          source={
            Object {
              "uri": "C",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              },
              Object {
                "resizeMode": "contain",
              },
            ]
          }
          virtual={undefined}
        />
      </View>
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": null,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 1,
            "marginLeft": 15,
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          C
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          numberOfLines={1}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          C
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "stretch",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 0,
            "paddingBottom": 10,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          style={
            Object {
              "alignSelf": null,
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "200",
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          2018-01-03
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      avatar={true}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 15,
          "paddingBottom": null,
          "paddingRight": null,
          "paddingTop": null,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 0,
            "flexDirection": "row",
          }
        }
        virtual={undefined}
      >
        <Image
          source={
            Object {
              "uri": "D",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              },
              Object {
                "resizeMode": "contain",
              },
            ]
          }
          virtual={undefined}
        />
      </View>
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": null,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 1,
            "marginLeft": 15,
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          D
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          numberOfLines={1}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          D
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "stretch",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 0,
            "paddingBottom": 10,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          style={
            Object {
              "alignSelf": null,
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "200",
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          2018-01-04
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      avatar={true}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 15,
          "paddingBottom": null,
          "paddingRight": null,
          "paddingTop": null,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 0,
            "flexDirection": "row",
          }
        }
        virtual={undefined}
      >
        <Image
          source={
            Object {
              "uri": "E",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              },
              Object {
                "resizeMode": "contain",
              },
            ]
          }
          virtual={undefined}
        />
      </View>
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": null,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 1,
            "marginLeft": 15,
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          E
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          numberOfLines={1}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginLeft": null,
              "marginRight": 10,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          E
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "stretch",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 0,
            "paddingBottom": 10,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          note={true}
          style={
            Object {
              "alignSelf": null,
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "200",
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          2018-01-05
        </Text>
      </View>
    </View>
  </View>
</View>
`;
